framework/OutStreams/PrintEntity.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 2, 2021
+
+@author: talbpaul
+"""
+import os
+import matplotlib
+
+from utils import utils, InputTypes
+from .OutStreamEntity import OutStreamEntity
+from .PrintInterfaces import factory as interfaceFactory
+
+
+class Print(OutStreamEntity):
+  """
+    Handler for Plot implementations
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    okTypes = list(interfaceFactory.knownTypes())
+    okEnum = InputTypes.makeEnumType('OutStreamPrint', 'OutStreamPrintType', okTypes)
+    spec.addParam('subType', required=False, param_type=okEnum, descr=r"""Type of OutStream Print to generate.""")
+    # TODO add specs depending on the one chosen, not all of them!
+    # FIXME the GeneralPlot has a vast need for converting to input specs. Until then,
+    #       we cannot strictly check anything related to it.
+    spec.strictMode = False
+    for name in okTypes:
+      printer = interfaceFactory.returnClass(name)
+      subSpecs = printer.getInputSpecification()
+      spec.mergeSub(subSpecs)
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'PrintEntity'
+    self._printer = None            # implemention, inheriting from interface
+
+  def _readMoreXML(self, xml):
```

Comment:
I suggest to move this method to Print class that are using this method. Not in the PrintEntity base class.

Reply:
Unfortunate the `BaseType` classifies the `OutStreams` as an XML user because its factory doesn't register the `OutStreams` module as returning the InputParameters. To return inputParameters requires some additional mechanics added, such as a distinct InputParams collector class to return the input specs.

This cannot currently be added, since the GeneralPlot does not use the InputParams, so the user-provided XML cannot be parsed correctly. I did try to work around this, but since GeneralPlot requires the XML, we need the BaseEntity to call the `_readMoreXML` method on `PlotEntity`, not the `handleInput` method.